src/lammpsio/data.py

```diff
@@ -100,13 +108,11 @@ def create(cls, filename, snapshot, atom_style=None):
             If all masses are not the same for a given type.
 
         """
-        # extract number of types
-        num_types = numpy.amax(numpy.unique(snapshot.typeid))
 
         # extract mass by type
         if snapshot.has_mass():
-            masses = numpy.empty(num_types)
-            for i in range(num_types):
+            masses = numpy.empty(snapshot.num_types)
```

Comment:
We should be careful here... if a type exists but not the particle, `empty` will leave the value of the mass for that type undefined. Maybe we should fill this with `np.nan` initially so that the mass is undefined? Or the LAMMPS default if it has a default (check docs)?

Reply:
Use `numpy.ones`